src/sentry/utils/services.py

```diff
@@ -465,6 +470,64 @@ def setup(self) -> None:
             backend.setup()
 
 
+KeyFetch = Callable[[Context, str, Mapping[str, Any]], str | int]
+
+
+def make_writebehind_selector(
+    *, option_name: str, key_fetch: KeyFetch, move_to: str, move_from: str
+) -> Selector:
+    """
+    Generates a selector_func that will do write-behind delegation
+
+    The provided option_name is expected to have values between -1 and 1
+
+    -1.0 - 0.01 The move_from will be primary, while move_to will increasingly be added as a secondary.
+    At 0.0 - Only move_from will be used.
+    0.01 - 1.0 The move_to will increasingly be used as primary.
+
+    The `key_fetch` function gets the parameters expected by `Selector` and
+    is expected to return a consistent str|int that will be hashed for consistent
+    rollouts. If no consistent key exists you can use random number generation.
+
+    Example:
+
+    selector = make_writebehind_selector(
+        option_name="feature.rollout",
+        move_to="new",
+        move_from="old",
+        key_fetch=lambda *args: "a-consistent-key",
+    )
+    """
+
+    def selector(context: Context, method: str, callargs: Mapping[str, Any]) -> list[str]:
+        rollout_rate = options.get(option_name)
```

Comment:
Wondering if it would be helpful at all for this to register the option automatically?

Reply:
I'm torn on auto-registering options. It makes building new rollouts easier, as options are created automatically but also hides the source of option definitions.